modules/page-memory/src/test/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/CheckpointTimeoutLockTest.java

```diff
@@ -407,4 +417,46 @@ private Checkpointer newCheckpointer(Thread runner, CompletableFuture<?> future)
 
         return checkpointer;
     }
+
+    @Test
+    void testCheckpointReadLockMetrics() throws Exception {
+        CheckpointMetricSource metricSource = new CheckpointMetricSource("test");
+        CheckpointReadWriteLockMetrics metrics = new CheckpointReadWriteLockMetrics(metricSource);
+        CheckpointReadWriteLock readWriteLock = newReadWriteLock(metrics);
+
+        timeoutLock = new CheckpointTimeoutLock(
+                readWriteLock,
+                10_000,
+                () -> NOT_REQUIRED,
+                mock(Checkpointer.class),
+                mock(FailureManager.class)
+        );
+
+        timeoutLock.start();
+
+        try {
+            // Verify metrics start at zero
+            assertTrue(Arrays.stream(metrics.readLockAcquisitionTime().value()).allMatch(it -> it == 0L));
```

Comment:
Is there a Hamcrest matcher that makes sure that every element satisfies a condition? Such an assertion would give more information on failure.

I would suggest to do the same with the assertions that follow

Reply:
I changed this assertion to be an explicit check of sum of all the buckets hits, this way it will be more fair and will hopefully clarify better the intent of the check (in the javadoc)